packages/swapper/src/swappers/zrx/zrxExecuteTrade/zrxExecuteTrade.ts

```diff
@@ -1,40 +1,23 @@
 import { numberToHex } from 'web3-utils'
 
-import { ExecuteTradeInput, SwapError, TradeResult } from '../../../api'
+import { ExecuteTradeInput, SwapError, SwapErrorTypes, TradeResult } from '../../../api'
+import { bnOrZero } from '../utils/bignumber'
 import { ZrxSwapperDeps } from '../ZrxSwapper'
 
 export async function zrxExecuteTrade(
   { adapterManager }: ZrxSwapperDeps,
   { trade, wallet }: ExecuteTradeInput<'eip155:1'>
 ): Promise<TradeResult> {
   const { sellAsset } = trade
-
-  if (!sellAsset.symbol) {
-    throw new SwapError('ZrxSwapper:ZrxExecuteTrade sellAssetSymbol is required')
-  }
-
-  if (!trade.sellAssetAccountId) {
-    throw new SwapError('ZrxSwapper:ZrxExecuteTrade sellAssetAccountId is required')
-  }
-
-  if (!trade.sellAmount) {
-    throw new SwapError('ZrxSwapper:ZrxExecuteTrade sellAmount is required')
-  }
-
-  if (!trade.depositAddress) {
-    throw new SwapError('ZrxSwapper:ZrxExecuteTrade depositAddress is required')
-  }
-
-  // value is 0 for erc20s
-  const value = sellAsset.symbol === 'ETH' ? trade.sellAmount : '0'
-  const adapter = adapterManager.byChain(sellAsset.chain)
-  const bip44Params = adapter.buildBIP44Params({
-    accountNumber: Number(trade.sellAssetAccountId)
-  })
-
-  let buildTxResponse, signedTx, txid
   try {
-    buildTxResponse = await adapter.buildSendTransaction({
+    // value is 0 for erc20s
+    const value = sellAsset.assetId === 'eip155:1/slip44:60' ? trade.sellAmount : '0'
+    const adapter = await adapterManager.byChainId(sellAsset.chainId)
+    const bip44Params = adapter.buildBIP44Params({
+      accountNumber: bnOrZero(trade.sellAssetAccountId).toNumber()
```

Comment:
separate ticket to rename this field on the trade type from `sellAssetAccountId` -> `sellAssetAccountNumber`

Reply:
https://github.com/shapeshift/lib/issues/653